src/celestia/celx_internal.h

```diff
@@ -149,8 +308,41 @@ class CelxLua
                         const char* errorMessage = "Boolean argument expected",
                         bool defaultValue = false);
 
-    LuaState* getLuaStateObject();
+    template<typename T> T *safeGetUserData(int index = 0,
+                                            FatalErrors errors = AllErrors,
+                                            const char *errorMessage = "User data expected")
+    {
+        if (!safeIsValid(index))  return nullptr;
+        if (!isUserData(index))
+        {
+            if (errors & WrongType)
+                doError(errorMessage);
+            return nullptr;
+        }
+        return reinterpret_cast<T*>(lua_touserdata(m_lua, index));
```

Comment:
There are places with wrong style. Better:
```
if (isUserData(index))
    return reinterpret_cast<T*>(lua_touserdata(m_lua, index));

if (errors & WrongType)
    doError(errorMessage);

return nullptr;
```
Such code is much easier to understand.